### PR TITLE
PR: Added Matlab files import tests

### LINKED ISSUE #4076: Add tests to verify that importing .mat files is working
Import of .mat file after upgrade to v3.1.2

files saved with 3.1.2 can be opened and saved in 3.1.0. 

image

### PR DESCRIPTION
**Pull request by andfoy**
Fixes #4076
This test allows to confirm and validate the main datatypes defined in MATLAB, including:
* Scalar numbers
* Vectors/Matrices
* Structs
* Strings
* Cell Arrays
* Complex values

### CODE REVIEW

**Comment by goanpeca** [comment]
Lookign good :-), ping when its done for review

**Comment by andfoy** [comment]
@goanpeca This PR is ready to go!
PS: There's an unrelated error to this PR while testing on Travis

**Comment by ccordoba12** [comment]
Yep, our Travis tests are failing because of a new release of Jedi. I'll fix it right away ;-)

File: spyder/utils/tests/test_iofuncs.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    D = file_s['D'].item()
+    E = file_s['E']
+    return {'A':A, 'B':B, 'C':C, 'D':D, 'E':E}
+
+def test_matlab_import(real_values):
+    path = os.path.join(__location__, 'data.mat')
+    inf, _ = io.load_matlab(path)
+    valid = True
+    for var in sorted(real_values.keys()):
+        valid = valid and np.sum(real_values[var] == inf[var])
```
Comment (ccordoba12) [inline spyder/utils/tests/test_iofuncs.py]: Why are you using `np.sum` here?

File: spyder/utils/tests/test_iofuncs.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+"""
+Tests for iofuncs.py
+"""
+
+import os
+import pytest
+import numpy as np
+import spyder.utils.iofuncs as io
+
+__location__ = os.path.realpath(os.path.join(os.getcwd(),
```
Comment (ccordoba12) [inline spyder/utils/tests/test_iofuncs.py]: Make this a constant, i.e. `LOCATION`

File: spyder/utils/tests/test_iofuncs.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    path = os.path.join(__location__, 'numpy_data.npz')
+    file_s = np.load(path)
+    A = file_s['A'].item()
+    B = file_s['B']
+    C = file_s['C']
+    D = file_s['D'].item()
+    E = file_s['E']
+    return {'A':A, 'B':B, 'C':C, 'D':D, 'E':E}
+
+def test_matlab_import(real_values):
```
Comment (ccordoba12) [inline spyder/utils/tests/test_iofuncs.py]: Please add a docstring to this test explaining how it works, i.e what are the contents of data.mat and numpy_data.npz, and what's the test expected result.

File: spyder/utils/tests/test_iofuncs.py [hunk 4]
```diff
@@ -13,13 +13,25 @@
 import numpy as np
 import spyder.utils.iofuncs as io
 
-__location__ = os.path.realpath(os.path.join(os.getcwd(),
+LOCATION = os.path.realpath(os.path.join(os.getcwd(),
                                              os.path.dirname(__file__)))
```
Comment (ccordoba12) [inline spyder/utils/tests/test_iofuncs.py]: Please align this line correctly to be at the same level of `os.getcwd()` above.

File: spyder/utils/tests/test_iofuncs.py [hunk 5]
```diff
@@ -13,13 +13,25 @@
...
 
 
 @pytest.fixture
 def real_values():
-    path = os.path.join(__location__, 'numpy_data.npz')
+    """
+    Loads a Numpy pickled file.
```
Comment (ccordoba12) [inline spyder/utils/tests/test_iofuncs.py]: Loads -> Load

File: spyder/utils/tests/test_iofuncs.py [hunk 6]
```diff
@@ -13,13 +13,25 @@
...
+
+    The file numpy_data.npz contains six variables, each one represents the
+    expected test values after a manual conversion of the same variables
+    defined and evaluated in MATLAB. The manual type conversion was done
+    over several variable types, such as: Matrices/Vectors, Scalar and
+    Complex numbers, Structs, Strings and Cell Arrays. The set of variables
+    was defined to allow and test the deep conversion of a compound type,
+    i.e., A struct that contains other types that need to be converted,
```
Comment (ccordoba12) [inline spyder/utils/tests/test_iofuncs.py]: typo: *i.e. a struct*

File: spyder/utils/tests/test_iofuncs.py [hunk 7]
```diff
@@ -29,7 +41,16 @@ def real_values():
     return {'A':A, 'B':B, 'C':C, 'D':D, 'E':E}
 
 def test_matlab_import(real_values):
-    path = os.path.join(__location__, 'data.mat')
+    """
+    Tests the automatic conversion and import of variables from MATLAB.
+
+    This test loads a file stored in MATLAB, the variables defined are
+    equivalent to the manually converted values done over Numpy. This test
+    allows to evaluate the function which processes the conversion automa
```
Comment (ccordoba12) [inline spyder/utils/tests/test_iofuncs.py]: automa -> automa-

**Pull request merged**